docs/basic-component.md

```diff
@@ -0,0 +1,247 @@
+import classnames from "classnames"
+import styles from "./PancakeStack.scss"
+
+export interface PancakeStackProps extends Omit<HTMLAttributes<HTMLDivElement>, "className"> {
+  children: React.ReactNode
+  classNameOverride?: string
+  isBooleanProp: boolean
+  hasOptionalBooleanProp?: boolean
+  onCustomFunction: () => void
+}
+
+export const PancakeStack: React.VFC<PancakeStackProps> = ({
+  children,
+  classNameOverride,
+  isBooleanProp,
+  hasOptionalBooleanProp = false,
+  onCustomFunction,
+  ...props
+}) => {
+  const [hasSyrup, setHasSyrup] = useState<boolean>(false)
+
+  const handleCustomFunction = () => {
+    onCustomFunction()
+    setHasSyrup(!hasSyrup)
+    return true
+  }
+
+  return (
+    <div
+      className={classnames(styles.pancakeStack, classNameOverride, {
+        [styles.someClass]: isBooleanProp,
+      })}
+      onSomething={handleCustomFunction}
+      {...props}
+    >
+      {children}
+      {doSomething(hasOptionalBooleanProp) && <SubComponent />}
+    </div>
+  );
+}
+
+PancakeStack.displayName = "PancakeStack"
+```
+
+### Props
+
+```tsx
+import { HTMLAttributes } from "react"
+
+export interface PancakeStackProps extends Omit<HTMLAttributes<HTMLDivElement>, "className"> {
+  children: React.ReactNode
+  classNameOverride?: string
+  isBooleanProp: boolean
+  hasOptionalBooleanProp?: boolean
+  onCustomFunction: () => void
+}
+```
+
+- Create and export an `interface` for your props in the format of `{PascalComponentName}Props`
+  - You can restrict a prop's type signature by redeclaring it (eg. changing a prop from `string | undefined` to `string`)
+  - If you need to change the type signature of an extended prop, ensure you `Omit` it otherwise it will become `never` (eg. changing a prop from `string | undefined` to `number`)
+- `Omit` `"className"` as we re-add it with an alias (`classNameOverride`)
+- Extend the native attributes of the closest HTML element of your component (eg. `<section>` will use `<div>` attributes)
+
+```tsx
+// Extending <section>
+import { HTMLAttributes } from 'react'
+export type SectionProps = Omit<HTMLAttributes<HTMLDivElement>, "className">
+
+// Extending <button>
+import { ButtonHTMLAttributes } from 'react'
+export type ButtonProps = Omit<ButtonHTMLAttributes<HTMLButtonElement>, "className">
+```
+
+- If your component is mainly extending another component, extend the props of that component and spread the props
+  - This will prevent the need for updating the consuming component with any new props in the child
+  - You don't need to extend HTML attributes as they would come from the child props (this also ensures consistency)
+
+```tsx
+export interface NewComponentProps extends PancakeStackProps {}
+
+export const NewComponent: React.VFC<NewComponentProps> = ({ ...props }) => <PancakeStack {...props} />
+```
+
+- Declare the `children` prop if you require it
+  - Usually it is `React.ReactNode`, however you can customise it to your needs
```

Comment:
Might be helpful to have this to reference:
https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement